fairroot/online/steer/FairRunOnline.cxx

```diff
@@ -209,60 +208,24 @@ void FairRunOnline::Init()
     fRootManager->WriteFileHeader(fFileHeader);
 }
 
-void FairRunOnline::InitContainers()
-{
-    fRtdb = GetRuntimeDb();
-    FairBaseParSet* par = static_cast<FairBaseParSet*>(fRtdb->getContainer("FairBaseParSet"));
-    LOG(info) << "FairRunOnline::InitContainers: par = " << par;
-    if (nullptr == par)
-        LOG(warn) << "FairRunOnline::InitContainers: no  'FairBaseParSet' container !";
-
-    if (par) {
-        fEvtHeader = static_cast<FairEventHeader*>(fRootManager->GetObject("EventHeader."));
-
-        fRunId = GetEvtHeaderRunId();
-
-        // Copy the Event Header Info to Output
-        fEvtHeader->Register();
-
-        // Init the containers in Tasks
-        fRtdb->initContainers(fRunId);
-        fTask->ReInitTask();
-        //    fTask->SetParTask();
-        fRtdb->initContainers(fRunId);
-        //     if (gGeoManager==0) {
-        //       par->GetGeometry();
-        //     }
-    } else {
-        // --- Get event header from Run
-        //      fEvtHeader = dynamic_cast<FairEventHeader*> (FairRunOnline::Instance()->GetEventHeade
-        GetEventHeader();
-        if (!fEvtHeader) {
-            LOG(fatal) << "FairRunOnline::InitContainers:No event header in run!";
-            return;
-        }
-        LOG(info) << "FairRunOnline::InitContainers: event header at " << fEvtHeader;
-        fRootManager->Register("EventHeader.", "Event", fEvtHeader, kTRUE);
-    }
-}
-
 Int_t FairRunOnline::EventLoop()
 {
     gSystem->IgnoreInterrupt();
     signal(SIGINT, handler_ctrlc);
 
     FillEventHeader();
     auto const tmpId = GetEvtHeaderRunId();
-
     if (tmpId != fRunId) {
-        LOG(info) << "FairRunOnline::EventLoop() Call Reinit due to changed RunID (from " << fRunId << " to " << tmpId;
-        fRunId = tmpId;
-        Reinit(fRunId);
-        if (!GetSource()->ReInitUnpackers()) {
-            LOG(fatal) << "FairRunOnline->EventLoop() ReInitUnpackers() failed!";
-            return 1;
+        if (!fStatic) {
+            LOG(info) << "FairRunOnline::EventLoop() Call Reinit due to changed RunID (from " << fRunId << " to "
+                      << tmpId << ")";
+            Reinit(tmpId);
+            fTask->ReInitTask();
+        } else {
+            LOG(info) << "FairRunOnline::EventLoop() Changed RunID (from " << fRunId << " to " << tmpId
+                      << "). ReInit not called because initialisation is static.";
         }
-        fTask->ReInitTask();
+        fRunId = tmpId;
```

Comment:
`fRunId is set quite late in your change. So for example `fTask->ReInitTask` might see the old fRunId?

FairRunAna sets it earlier:

https://github.com/FairRootGroup/FairRoot/blob/76fc0d676477f188843af073e74fa2ffd789db09/fairroot/base/steer/FairRunAna.cxx#L347-L351